connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/PlanGenerationTestSuite.scala

```diff
@@ -698,6 +698,11 @@ class PlanGenerationTestSuite
     simple.distinct()
   }
 
+  test("select collated string") {
+    val schema = StructType(StructField("s", StringType(1)) :: Nil)
```

Comment:
@dbatomic just noticed that we expose collation id in the public API. I think it's better to only expose the collation name in the public API, and make the `def apply` that takes collation id be `private[sql]` so that it's only allowed to be used within the Spark codebase.